build.gradle

```diff
@@ -37,9 +34,17 @@ apply from: "$rootDir/gradle/idea.gradle"
 apply from: "$rootDir/gradle/tests.gradle"
 
 // The task for running example application from the command line.
-//#docregion run-task
 task sayHello(type: JavaExec) {
     main = 'io.spine.helloworld.Example'
     classpath = sourceSets.main.runtimeClasspath
 }
-//#enddocregion run-task
+
+// The task for composite builds to build this example.
+//
+// This build task is not a part of the example. It is used by the composite build of
+// the `spine.io` project which includes this project for embedding the source code of
+// this example into the site.
+//
+task buildAll(type: GradleBuild) {
```

Comment:
How is this `buildAll` different from just `build`?

Also, If I get the intention, we want to have the same similar task for **all** examples so that the `spine.io` project is able to build them as part of its own build. Why not have a plugin for this? It'll allow simplifying the build, having the same consistent build task/strategy for all examples.

Reply:
Here's how these parts are gathered in the `spine.io` build:

```groovy
/**
 * Builds all included projects via depending on the top-level "buildAll" tasks
 * declared in these proejcts.
 *
 * @see https://discuss.gradle.org/t/defining-a-composite-build-only-to-build-all-subprojects/25070/6
 * @see https://github.com/AlexMAS/gradle-composite-build-example
 * @see https://docs.gradle.org/current/userguide/composite_builds.html
 */
tasks.register("buildAll")  {
    dependsOn(gradle.includedBuilds.map { it.task(":buildAll") })
}
```

Multi-module builds do not have the `build` task at the top. `buildAll` is a kind of synthetic task we need to be able to build all the examples. See the links in the docs for the description of the problem and the suggested solution.

Reply:
> How is this buildAll different from just build?

In the single-module project like this one, there's no difference. But we need to have a task with the common name (`buildAll`) which would be used by the gathering project.